.github/CONTRIBUTING.md

```diff
@@ -10,6 +10,23 @@ First time contributing to Homebrew? Read our [Code of Conduct](https://github.c
 
 ### Propose a feature
 
-* Please create a pull request with your change proposal in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md).
+* create a pull request in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md)
+
+### Contribute code
+
+* there are no universally applicable rules (a lot depend on the scope of your change)
+* before you start coding, consider [proposing a feature](#propose-a-feature) (see above) or discussing the idea in an issue
+* to learn how to structure your commits, consult the Git history of modified files
+* [open a pull request](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) on this repository
+
+### Improve help for `brew <command>`
```

Comment:
Agreed. I just wanted to get this documented so it's easier for people to discover (or get linked to). Wasn't sure where to best put this, but a separate document seems reasonable.